### PR TITLE
Fixes pride mirror centcom exploit, teeny bit of code improvement

### LINKED ISSUE #31427: Mirror of pride can be brought to and used at exploitable positions.
Title,
Clever use of meteor-shot and a hand-tele is all it takes to bring one of these mirrors to other z-levels, where it can be used to drop down on places like Centcom and The City of Cogs.
Suffice to say holes in general are bad, especially these kind.

### PR DESCRIPTION
**Pull request by ghost**
fixes #31427

### CODE REVIEW

**Comment by Armhulen** [comment]
i know, now it's an instant kill. I think the whole point of the ruin was to bring a jaunter anyways, instead of being forced to pick a spaceworthy species or die. That's dumb!

**Comment by praisenarsie** [comment]
wait what exploit is this fixing

File: code/modules/ruins/objects_and_mobs/sin_ruins.dm [hunk 1]
```diff
@@ -101,8 +101,8 @@
 	user.visible_message("<span class='danger'><B>The ground splits beneath [user] as [user.p_their()] hand leaves the mirror!</B></span>", \
 	"<span class='notice'>Perfect. Much better! Now <i>nobody</i> will be able to resist yo-</span>")
 	var/turf/T = get_turf(user)
-	T.ChangeTurf(/turf/open/chasm/straight_down)
-	var/turf/open/chasm/straight_down/C = T
+	T.ChangeTurf(/turf/open/chasm)
+	var/turf/open/chasm/C = T
```
Comment (Cyberboss) [inline code/modules/ruins/objects_and_mobs/sin_ruins.dm]: var/turf/open/chasm/C = T.ChangeTurf(/turf/open/chasm)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
@praisenarsie 
1. build meteor shot
2. get teleporter
3.place teleporter right next to pride mirror
4. meteor shot pride mirror into teleporter
5. if the pride mirror is on the station you now have access to centcom

**Comment by Arianya** [comment]
Just make the mirror immune to meteor shot or unteleporteable.

:-1: on making it a true chasm

**Comment by Xhuis** [comment]
I agree with the instakill implementation over what we have now. I do wonder if it'd be a better idea to make it drop to a fixed z instead, though.

**Comment by ShizCalev** [comment]
image

How I was going to handle it if #31398 was actually fixed. :(

**Comment by JJRcop** [comment]
Wrong way to do this, the real issue is straight_down is taking you to centcom. Don't we have a multiz list already?

**Comment by ShizCalev** [comment]
z_is_planet & station_z_levels

**Comment by Armhulen** [comment]
But conceptually teleporting you to space for changing your name is dumb. You're FORCED to pick a space worthy race or you die when this should just be a better mirror. Jaunter to counter or come home cryin

**Comment by Arianya** [comment]
Or you grab a space suit.

**Comment by Armhulen** [comment]
Son of a bitch I NEED GBP AAAAAAAAAAAAAAAAH

**Pull request closed without merging**